### DISCUSSION TITLE
Github Pages as PWA app for Play Market

### DISCUSSION THREAD

**Discussion by nurgeldiserikbay** [community/community#114770] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Good afternoon. Can I use GitHub Pages as a PWA application to upload to third party app stores. In this case, on the play market. The application is a static page, but there will be an advertising banner in the application itself? Thanks for the answer

**Comment by benogoulart** [comment]
Hi,

Yes, you can host a **Progressive Web App (PWA)** on GitHub Pages as long as it consists of static assets (HTML, CSS, JavaScript, images, etc.).

In general:

- ✅ GitHub Pages can host static PWAs.
- ✅ You can include advertising banners, provided they comply with GitHub's Terms of Service and the policies of the app store where you publish the app.
- ✅ You can package the PWA for distribution on third-party app stores, including Google Play, as long as you meet that store's requirements.

Keep in mind that GitHub Pages only provides static hosting, so if your application requires server-side functionality (such as authentication, databases, or APIs), you'll need to use an external backend service.

If your goal is to publish the PWA on Google Play, you'll also need to ensure it meets Google Play's current requirements for web-based apps.

Hope this helps!

**Comment by Mehrab-Javadzadeh** [comment]
I do not see any reason why you cannot host your PWA on GitHub Pages and then package it for distribution via a third-party app store, as long as you adhere to the terms of service of both GitHub and the app store you are using.
Given that your app is a static website, GitHub Pages seems like a legitimate choice. The presence of an advertisement banner in your app does not seem to be a problem, as long as you are adhering to the guidelines of the app store you are distributing your app via and your usage of GitHub Pages does not go against the acceptable use guidelines of GitHub.
My main concern would be to find out what the latest guidelines for PWAs or wrapped web apps from the Play Store are, since such guidelines can vary.

It would also be helpful if someone who has successfully distributed a PWA hosted on GitHub Pages through Google Play could provide feedback.
